recovery/main.cpp

```diff
@@ -81,10 +81,31 @@ QString findRecoveryDrive()
 
     foreach (QString devname, list)
     {
-        /* Only search first partition and partitionless devices. Skip virtual devices (such as ramdisk) */
-        if ((devname.right(1).at(0).isDigit() && !devname.endsWith("1"))
-                || QFile::symLinkTarget("/sys/class/block/"+devname).contains("/devices/virtual/"))
+        if (QFile::symLinkTarget("/sys/class/block/"+devname).contains("/devices/virtual/") )
+        {   //Skip if partition is virtual.
             continue;
+        }
+
+        int partitionNr = 0;
+        QRegExp partnrRx("([0-9]+)$");
+        if (partnrRx.indexIn(devname) != -1)
+        {   //Does devname end in 1 or more digits? Yes->Read it
+            partitionNr = partnrRx.cap(1).toInt();
+
+            if (partitionNr>1)
+            {   //Skip if partition Number >1
+                continue;
+            }
+        }
+
+        if (partitionNr == 0)
```

Comment:
I guess technically you could have `int partitionNr = -1;` on line 89, and then you could do `if ( ((partitionNr == -1) && list.contains(devname+"1")) || ((partitionNr == 0) && list.contains(devname+"p1")) )` (or spilt it into multiple `if`s), but maybe that's an optimisation that it's not worth making? :man_shrugging:

Reply:
Yes, you could, but I don't think it's worth it.
When I tested on Ubuntu, it also found /dev/sr0 which got scanned, but I'm happy with it as is.